great_expectations/core/datasource_dict.py

```diff
@@ -210,12 +226,15 @@ def __contains__(self, name: object) -> bool:
             return False
 
     @override
-    def __setitem__(self, name: str, ds: FluentDatasource | BaseDatasource) -> None:
+    def set_datasource(
+        self, name: str, ds: FluentDatasource | BaseDatasource
+    ) -> FluentDatasource | BaseDatasource | None:
         self.data[name] = ds
 
         # FDS do not use stores
         if not isinstance(ds, FluentDatasource):
-            super().__setitem__(name, ds)
+            return super().set_datasource(name, ds)
```

Comment:
Nit - Could we use key-word args

Reply:
Sure. Addressed in https://github.com/great-expectations/great_expectations/pull/8773/commits/864c95207ad75eca04638f3d16fbf80fe1ec2695